pkg/controller/restore-operator/http.go

```diff
@@ -98,6 +99,24 @@ func (r *Restore) serveBackup(w http.ResponseWriter, req *http.Request) error {
 
 		backupReader = reader.NewS3Reader(s3Cli.S3)
 		path = s3RestoreSource.Path
+	case api.BackupStorageTypeABS:
+		restoreSource := cr.Spec.RestoreSource
+		if restoreSource.ABS == nil {
+			return errors.New("empty abs restore source")
+		}
+		absRestoreSource := restoreSource.ABS
+		if len(absRestoreSource.ABSSecret) == 0 || len(absRestoreSource.Path) == 0 {
+			return errors.New("invalid abs restore source field (spec.abs), must specify all required subfields")
+		}
+
+		absCli, err := absfactory.NewClientFromSecret(r.kubecli, r.namespace, absRestoreSource.ABSSecret)
+		if err != nil {
+			return fmt.Errorf("failed to create ABS client: %v", err)
+		}
+		// Nothing to Close for absCli yet
```

Comment:
when should absCli be closed?

Reply:
it is simply an object. We don't need to explicit call Close. It would be GC when no reference :)

Reply:
@rjtsdl sometime a client can create gorutines that must be called closed to avoid any go routines leaks. not sure if that's the case for abd client.

Reply:
absCli doesn't do that. :)

Reply:
good to know.